pkg/init/backend/interactive.go

```diff
@@ -133,8 +134,21 @@ func (o *InteractiveBackend) PersonalizeName(devfile parser.DevfileObj, flags ma
 	if name == "" {
 		return "", fmt.Errorf("unable to detect the name")
 	}
-
-	return o.askerClient.AskName(name)
+	var userReturnedName string
+	userReturnedName, err = o.askerClient.AskName(name)
+	if err != nil {
+		return "", err
+	}
+	// keep asking the name until the user enters a valid name
+	for validK8sNameErr := dfutil.ValidateK8sResourceName("name", userReturnedName); validK8sNameErr != nil; {
+		log.Error(validK8sNameErr)
+		userReturnedName, err = o.askerClient.AskName(name)
+		if err != nil {
+			return "", err
+		}
+		validK8sNameErr = dfutil.ValidateK8sResourceName("name", userReturnedName)
+	}
```

Comment:
You are repeating the AskName related code, and executing `ValidateK8sResourceName` twice on every loop.

You can do something similar to:

```
for {
   userReturnedName, err = o.askerClient.AskName(name)

  validK8sNameErr = dfutil.ValidateK8sResourceName("name", userReturnedName)
  if validK8sNameErr == nil {
    break
  }
}
```

Reply:
>executing ValidateK8sResourceName twice on every loop

I don't think it's executed twice on every loop, but what you've suggested here looks like a better solution.

Reply:
oh yes you are right, it is called only once. But the code is still repeated